### PR TITLE
Refactor/people details

### PR DESCRIPTION
**Pull request by Hagar-Gomaa**

### CODE REVIEW

File: core/bases/src/main/java/com/chocolatecake/util/BindingAdapters.kt [hunk 1]
```diff
@@ -81,4 +82,24 @@ fun <T> View.showWhenError(list: List<T>?){
...
+fun TextView.convertGenderText(gender: String?) {
+    text = if (gender != null) {
+        when (gender) {
+            "1" -> "female"
+            "2" -> "male"
+            else -> ""
+        }
+    } else {
+        ""
+    }
```
Comment (Saif-Alhaider) [inline core/bases/src/main/java/com/chocolatecake/util/BindingAdapters.kt:97]: you could rewrite this in better way like this
```
text = when (gender) {
    "1" -> "female"
    "2" -> "male"
    else -> ""
}.takeIf { gender != null } ?: ""
```

Comment (Saif-Alhaider) [reply core/bases/src/main/java/com/chocolatecake/util/BindingAdapters.kt:97]: also don't make male and female hard codded values add them in string res

File: core/bases/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -24,4 +24,6 @@
     <string name="company">Company</string>
     <string name="apply">Apply</string>
     <string name="Popular_Movies">Popular Movies</string>
+    <string name="biography">Biography</string><string name="movies">Movies</string>
+    <string name="tv_shows">TV Shows</string>
```
Comment (Saif-Alhaider) [inline core/bases/src/main/res/values/strings.xml:28]: add new line for movise (format the code)

File: core/data/remote/src/main/java/com/chocolatecake/remote/response/TvShowsByPeopleResponse.kt [hunk 3]
```diff
@@ -0,0 +1,68 @@
+package com.chocolatecake.remote.response
+import com.google.gson.annotations.SerializedName
+
+data class TvShowsByPeopleResponse(
+
+    @field:SerializedName("cast")
```
Comment (Saif-Alhaider) [inline core/data/remote/src/main/java/com/chocolatecake/remote/response/TvShowsByPeopleResponse.kt:6]:
```suggestion
    @SerializedName("cast")
```
remove field

File: core/data/remote/src/main/java/com/chocolatecake/remote/response/MoviesByPeopleResponse.kt
Comment (Saif-Alhaider) [inline core/data/remote/src/main/java/com/chocolatecake/remote/response/MoviesByPeopleResponse.kt:1]: rename to MoviesByPersonResponse

File: core/data/remote/src/main/java/com/chocolatecake/remote/response/PeopleDetailsResponse.kt [hunk 4]
```diff
@@ -0,0 +1,48 @@
+package com.chocolatecake.remote.response
+
+import com.google.gson.annotations.SerializedName
+
+data class PeopleDetailsResponse(
+
+    @field:SerializedName("also_known_as")
```
Comment (Saif-Alhaider) [inline core/data/remote/src/main/java/com/chocolatecake/remote/response/PeopleDetailsResponse.kt:7]:
```suggestion
    @field:SerializedName("also_known_as")
```
remove field

File: core/data/remote/src/main/java/com/chocolatecake/remote/service/MovieService.kt [hunk 5]
```diff
@@ -74,6 +77,17 @@ interface MovieService {
     /// region popular people
     @GET("person/popular")
     suspend fun getPopularPeople(@Query("page") page: Int = 1): Response<DataWrapperResponse<PeopleRemoteDto>>
+
```
Comment (Saif-Alhaider) [inline core/data/remote/src/main/java/com/chocolatecake/remote/service/MovieService.kt:80]: remove empty line

File: core/data/repository/src/main/java/com/chocolatecake/repository/MovieRepositoryImpl.kt [hunk 6]
```diff
@@ -175,5 +183,17 @@ class MovieRepositoryImpl @Inject constructor(
         } catch (_: Throwable) {
         }
     }
+
+    override suspend fun getPerson(person_id: Int): PeopleDataEntity {
+        return domainPeopleDetailsMapper.map(wrapApiCall { movieService.getPerson(person_id) })
+    }
+
+    override suspend fun getMoviesByPerson(person_id: Int): List<MovieEntity> {
+        return domainMoviesByPeopleMapper.map( wrapApiCall { movieService.getMoviesByPerson(person_id)}.cast!!)
```
Comment (Saif-Alhaider) [inline core/data/repository/src/main/java/com/chocolatecake/repository/MovieRepositoryImpl.kt:192]:
```suggestion
val call = wrapApiCall { movieService.getMoviesByPerson(person_id)}.cast?.filterNotNull() ?: emptylist()
        return domainMoviesByPeopleMapper.map(call)
```
filter not null cast do not pass !! in map

File: core/domain/entities/src/main/java/com/chocolatecake/entities/PeopleDataEntity.kt
Comment (Saif-Alhaider) [inline core/domain/entities/src/main/java/com/chocolatecake/entities/PeopleDataEntity.kt:1]: rename to PeopleDetailsEntity

File: core/data/repository/src/main/java/com/chocolatecake/repository/mappers/domain/DomainPeopleDetailsMapper.kt [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+class DomainPeopleDetailsMapper @Inject constructor() :
+    Mapper<PeopleDetailsResponse, PeopleDataEntity> {
+    override fun map(input: PeopleDetailsResponse): PeopleDataEntity {
+        return PeopleDataEntity(
+            id = input.id ?: 0,
+            name = input.name ?: "",
+            imageUrl = (BuildConfig.IMAGE_BASE_PATH + input.profilePath) ?: "", placeOfBirth = input.placeOfBirth ?: "",
+            gender = input.gender.toString(),
+            acting = input.knownForDepartment.toString(), num_movies = "", biography = input.biography ?:""
```
Comment (Saif-Alhaider) [inline core/data/repository/src/main/java/com/chocolatecake/repository/mappers/domain/DomainPeopleDetailsMapper.kt:17]: format 
add new line for each attribute

File: core/data/repository/src/main/java/com/chocolatecake/repository/mappers/domain/DomainPeopleDetailsMapper.kt [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
```
Comment (Saif-Alhaider) [inline core/data/repository/src/main/java/com/chocolatecake/repository/mappers/domain/DomainPeopleDetailsMapper.kt:18]: empty line

File: core/data/repository/src/main/java/com/chocolatecake/repository/mappers/domain/movie/DomainMoviesByPeopleMapper.kt [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+class DomainMoviesByPeopleMapper @Inject constructor() : Mapper<CastItem?,MovieEntity> {
+    override fun map(input: CastItem?):MovieEntity {
+
+        return MovieEntity(
+            id = input?.id ?: 0, input?.title ?: "",
+            (BuildConfig.IMAGE_BASE_PATH + input?.posterPath) , rate = input?.voteAverage
+                ?: 0.0
+        )
+
```
Comment (Saif-Alhaider) [inline core/data/repository/src/main/java/com/chocolatecake/repository/mappers/domain/movie/DomainMoviesByPeopleMapper.kt:17]: empty line

File: core/data/repository/src/main/java/com/chocolatecake/repository/mappers/domain/movie/DomainMoviesByPeopleMapper.kt [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+import com.chocolatecake.repository.mappers.Mapper
+import javax.inject.Inject
+
+class DomainMoviesByPeopleMapper @Inject constructor() : Mapper<CastItem?,MovieEntity> {
+    override fun map(input: CastItem?):MovieEntity {
+
+        return MovieEntity(
+            id = input?.id ?: 0, input?.title ?: "",
+            (BuildConfig.IMAGE_BASE_PATH + input?.posterPath) , rate = input?.voteAverage
+                ?: 0.0
```
Comment (Saif-Alhaider) [inline core/data/repository/src/main/java/com/chocolatecake/repository/mappers/domain/movie/DomainMoviesByPeopleMapper.kt:15]: add new line for each attribute

File: core/data/repository/src/main/java/com/chocolatecake/repository/mappers/domain/movie/DomainTvShowsByPeopleMapper.kt [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+import com.chocolatecake.repository.mappers.Mapper
+import javax.inject.Inject
+
+class DomainTvShowsByPeopleMapper  @Inject constructor(): Mapper<TvShowsCastItem?, TvShowsEntity> {
+    override fun map(input: TvShowsCastItem?): TvShowsEntity{
+
+        return TvShowsEntity(
+            input?.id ?: 0, input?.name ?: "",
+            (BuildConfig.IMAGE_BASE_PATH + input?.posterPath) ?: "", rate = (input?.voteAverage
+                    ?: 0.0) as Double
```
Comment (Saif-Alhaider) [inline core/data/repository/src/main/java/com/chocolatecake/repository/mappers/domain/movie/DomainTvShowsByPeopleMapper.kt:15]: add new line for each attribute

File: core/data/repository/src/main/java/com/chocolatecake/repository/mappers/domain/movie/DomainTvShowsByPeopleMapper.kt [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+            )
+
+
```
Comment (Saif-Alhaider) [inline core/data/repository/src/main/java/com/chocolatecake/repository/mappers/domain/movie/DomainTvShowsByPeopleMapper.kt:18]: remove empty lines

File: core/domain/usecase/src/main/java/com/chocolatecake/repository/MovieRepository.kt [hunk 13]
```diff
@@ -34,5 +36,9 @@ interface MovieRepository {
 
     suspend fun getGenresMovies(): List<GenreEntity>
     suspend fun refreshGenres()
+    suspend fun getPerson( person_id: Int): PeopleDataEntity
+    suspend fun getMoviesByPerson( person_id: Int): List<MovieEntity>
+    suspend fun getTvShowsByPerson( person_id: Int): List<TvShowsEntity>
+
```
Comment (Saif-Alhaider) [inline core/domain/usecase/src/main/java/com/chocolatecake/repository/MovieRepository.kt:42]: remove empty line

File: core/domain/usecase/src/main/java/com/chocolatecake/usecase/people/GetMoviesByPeopleUseCase.kt
Comment (Saif-Alhaider) [inline core/domain/usecase/src/main/java/com/chocolatecake/usecase/people/GetMoviesByPeopleUseCase.kt:1]: rename to GetMoviesByPersonUseCase

File: core/domain/usecase/src/main/java/com/chocolatecake/usecase/people/GetTvShowsByPeopleUseCase.kt
Comment (Saif-Alhaider) [inline core/domain/usecase/src/main/java/com/chocolatecake/usecase/people/GetTvShowsByPeopleUseCase.kt:1]: rename to GetTvShowsByPersonUseCase

File: feature/home/presentation/home-ui/src/main/java/com/chocolatecake/ui/common/adapters/MediaVerticalAdapter.kt [hunk 14]
```diff
@@ -2,12 +2,12 @@ package com.chocolatecake.ui.common.adapters
...
 import com.chocolatecake.bases.BaseAdapter
 import com.chocolatecake.ui.home.R
-import com.chocolatecake.viewmodel.common.listener.MediaListener
 import com.chocolatecake.viewmodel.common.model.MediaVerticalUIState
+import com.chocolatecake.viewmodel.people.PeopleDetailsViewModel
 
 class MediaVerticalAdapter(
     list: List<MediaVerticalUIState>,
-    listener: MediaListener
+    listener: PeopleDetailsViewModel
```
Comment (Saif-Alhaider) [inline feature/home/presentation/home-ui/src/main/java/com/chocolatecake/ui/common/adapters/MediaVerticalAdapter.kt:10]: make PeopleDetailsViewModel implement MediaListener and return it to MediaListener

File: feature/home/presentation/home-ui/src/main/java/com/chocolatecake/ui/people/PeopleDetailsFragment.kt [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+                    }
+                }
+            }
+        }
+    }
+
+    override fun onEvent(event: PeopleDetailsUiEvent) {
+        when (event) {
+            PeopleDetailsUiEvent.BackNavigate -> findNavController().popBackStack()
+            else -> {}
```
Comment (Saif-Alhaider) [inline feature/home/presentation/home-ui/src/main/java/com/chocolatecake/ui/people/PeopleDetailsFragment.kt:62]: no need for else add all the events in when statement

File: feature/home/presentation/home-ui/src/main/java/com/chocolatecake/ui/people/PeopleDetailsFragment.kt [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+
+    private fun getData() {
+        collectLatest {
+            viewModel.state.collect { state ->
+                mediaVerticalAdapter.setItems(state.Movies)
+                mediaVerticalAdapterTvShows.setItems(state.TvShows)
+                if (state.onErrors.isNotEmpty()) {
+                    state.onErrors.last().let {
+                        showSnackBar(it)
+                    }
```
Comment (Saif-Alhaider) [inline feature/home/presentation/home-ui/src/main/java/com/chocolatecake/ui/people/PeopleDetailsFragment.kt:53]: when there is error it must display an error lottie with refresh button that is called in viewModel

File: feature/home/presentation/home-ui/src/main/java/com/chocolatecake/ui/people/PeopleDetailsFragment.kt [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+        }
+    }
+
+    override fun onEvent(event: PeopleDetailsUiEvent) {
+        when (event) {
+            PeopleDetailsUiEvent.BackNavigate -> findNavController().popBackStack()
+            else -> {}
+        }
+    }
+
```
Comment (Saif-Alhaider) [inline feature/home/presentation/home-ui/src/main/java/com/chocolatecake/ui/people/PeopleDetailsFragment.kt:65]: remove empty line

File: feature/home/presentation/home-ui/src/main/java/com/chocolatecake/ui/people/adapters/MoviesByPeopleAdapter.kt [hunk 18]
```diff
@@ -0,0 +1,10 @@
+package com.chocolatecake.ui.people.adapters
+
+import com.chocolatecake.bases.BaseAdapter
+import com.chocolatecake.ui.home.R
+import com.chocolatecake.viewmodel.people.PeopleDetailsListener
+import com.chocolatecake.viewmodel.people.PeopleDetailsUiState
+
+
+class MoviesByPeopleAdapter{
+}
```
Comment (Saif-Alhaider) [inline feature/home/presentation/home-ui/src/main/java/com/chocolatecake/ui/people/adapters/MoviesByPeopleAdapter.kt:10]: where is the implementation and why it does not inherit from the base adapter ??

File: feature/home/presentation/home-ui/src/main/java/com/chocolatecake/ui/people/adapters/TvShowsByPeopleAdapter.kt [hunk 19]
```diff
@@ -0,0 +1,11 @@
...
+
+import com.chocolatecake.bases.BaseAdapter
+import com.chocolatecake.ui.home.R
+import com.chocolatecake.viewmodel.people.PeopleDetailsListener
+import com.chocolatecake.viewmodel.people.PeopleDetailsUiState
+
+
+class TvShowsByPeopleAdapter{
+
+}
```
Comment (Saif-Alhaider) [inline feature/home/presentation/home-ui/src/main/java/com/chocolatecake/ui/people/adapters/TvShowsByPeopleAdapter.kt:11]: where is the implementation and why it does not inherit from the base adapter ??

File: feature/home/presentation/home-ui/src/main/res/layout/fragment_people_details.xml
Comment (Saif-Alhaider) [inline feature/home/presentation/home-ui/src/main/res/layout/fragment_people_details.xml:1]: root element should be CoordinatorLayout and the toolbar should be outide the scroll view 
look at movie details fragment and tv details fragment
ref: https://youtu.be/6UmHGn076To

File: feature/home/presentation/home-ui/src/main/res/layout/item_media_vertical.xml [hunk 20]
```diff
@@ -7,17 +7,12 @@
...
-        <variable
-            name="listener"
-            type="com.chocolatecake.viewmodel.common.listener.MediaListener" />
-    </data>
+   </data>
 
     <androidx.constraintlayout.widget.ConstraintLayout
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:onClick="@{() -> listener.onClickMedia(item.id)}"
```
Comment (Saif-Alhaider) [inline feature/home/presentation/home-ui/src/main/res/layout/item_media_vertical.xml:19]: why remove listener how can i click on the media ?

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsItem.kt
Comment (Saif-Alhaider) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsItem.kt:1]: i don't see any need for people details item if you are not doning nested rv

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsListener.kt [hunk 21]
```diff
@@ -0,0 +1,10 @@
+package com.chocolatecake.viewmodel.people
+
+import com.chocolatecake.bases.BaseInteractionListener
+
+interface PeopleDetailsListener : BaseInteractionListener {
+    fun onClickMovies(itemId: Int)
+    fun onClickTvShows(itemId: Int)
+    fun backNavigate()
+
+}
```
Comment (Saif-Alhaider) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsListener.kt:10]: make PeopleDetailsListener  implement media listener and then make the viewmodel implement PeopleDetailsListener

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsUiState.kt [hunk 22]
```diff
@@ -0,0 +1,35 @@
...
+//        val id: Int,
+//        val imageUrl: String,
+//        val rate: Double
+//    )
+//
+//    data class TvShowsUiState(
+//        val id: Int,
+//        val imageUrl: String,
+//        val rate: Double
+//    )
```
Comment (Saif-Alhaider) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsUiState.kt:24]: remove comments

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsUiEvent.kt [hunk 23]
```diff
@@ -0,0 +1,11 @@
+package com.chocolatecake.viewmodel.people
+
+import com.chocolatecake.viewmodel.home.HomeUiEvent
+
+sealed interface PeopleDetailsUiEvent {
+    data class ClickMovieEvent(val itemId: Int) : PeopleDetailsUiEvent
+    data class ClickTvShowsEvent(val itemId: Int) : PeopleDetailsUiEvent
+
+    object BackNavigate : PeopleDetailsUiEvent
+
```
Comment (Saif-Alhaider) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsUiEvent.kt:10]: if these functions are in the listener then there is no need for them as an events

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsUiState.kt [hunk 24]
```diff
@@ -0,0 +1,35 @@
+package com.chocolatecake.viewmodel.people
+
+import com.chocolatecake.viewmodel.common.model.MediaVerticalUIState
+
+
+data class PeopleDetailsUiState(
+    val peopleData :PeopleDataUiState =PeopleDataUiState(0,"","","","","","","") ,
```
Comment (Saif-Alhaider) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsUiState.kt:7]:
```suggestion
    val peopleData :PeopleDataUiState =PeopleDataUiState() ,
```
the ui state class must have already inital values

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsUiState.kt [hunk 25]
```diff
@@ -0,0 +1,35 @@
...
+//        val rate: Double
+//    )
+//
+//    data class TvShowsUiState(
+//        val id: Int,
+//        val imageUrl: String,
+//        val rate: Double
+//    )
+
+    data class PeopleDataUiState(
```
Comment (Saif-Alhaider) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsUiState.kt:26]: change name to PersonInfoUiState

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsUiState.kt [hunk 26]
```diff
@@ -0,0 +1,35 @@
...
+
+import com.chocolatecake.viewmodel.common.model.MediaVerticalUIState
+
+
+data class PeopleDetailsUiState(
+    val peopleData :PeopleDataUiState =PeopleDataUiState(0,"","","","","","","") ,
+    val Movies: List<MediaVerticalUIState> = emptyList(),
+    val TvShows: List<MediaVerticalUIState> = emptyList(),
+    val onErrors: List<String> = emptyList(),
+    val isLoading: Boolean = false
```
Comment (Saif-Alhaider) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsUiState.kt:11]: isLoading default value is true

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsUiState.kt (hunk 25 above)
Comment (Saif-Alhaider) [reply feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsUiState.kt:26]: PeopleDataUiState is only used in the data class PeopleDetailsUiState so add it in its body instead

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsUiState.kt [hunk 27]
```diff
@@ -0,0 +1,35 @@
+package com.chocolatecake.viewmodel.people
+
+import com.chocolatecake.viewmodel.common.model.MediaVerticalUIState
+
+
+data class PeopleDetailsUiState(
```
Comment (Saif-Alhaider) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsUiState.kt:6]: PersonDetailsUiState

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsViewModel.kt [hunk 28]
```diff
@@ -0,0 +1,130 @@
...
+        tryToExecute(
+            call =  {getMoviesByPeopleUseCase.invoke(personId)} ,
+            onSuccess = ::onSuccessGetMoviesByPeople,
+            onError = ::onErrorGetMoviesByPeople,
+            mapper = moviesByPeopleUiMapper
+        )
+    }
+
+
+    private fun onSuccessGetMoviesByPeople(moviesUiState: List<MediaVerticalUIState>) {
```
Comment (Saif-Alhaider) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsViewModel.kt:74]: should be List<MovieEntity>

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsUiState.kt [hunk 29]
```diff
@@ -0,0 +1,35 @@
...
+    val peopleData :PeopleDataUiState =PeopleDataUiState(0,"","","","","","","") ,
+    val Movies: List<MediaVerticalUIState> = emptyList(),
+    val TvShows: List<MediaVerticalUIState> = emptyList(),
```
Comment (Saif-Alhaider) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsUiState.kt:9]: variables first letter is small case

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsViewModel.kt [hunk 30]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+
+    override fun onClickMovies(itemId: Int) {
+        TODO("Not yet implemented")
+    }
+
+    override fun onClickTvShows(itemId: Int) {
+        TODO("Not yet implemented")
+    }
```
Comment (Saif-Alhaider) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsViewModel.kt:124]: where is the implementation ?

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsViewModel.kt [hunk 31]
```diff
@@ -0,0 +1,130 @@
...
+
+    override fun backNavigate() {
+        sendEvent(PeopleDetailsUiEvent.BackNavigate)
+
```
Comment (Saif-Alhaider) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsViewModel.kt:128]: remove empty line

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsViewModel.kt [hunk 32]
```diff
@@ -0,0 +1,130 @@
...
+    private fun onSuccessGetMoviesByPeople(moviesUiState: List<MediaVerticalUIState>) {
+        _state.update {
+            it.copy(
+                Movies = moviesUiState,
+                isLoading = false,
+            )
+        }
+        num_people_movies=moviesUiState.size.toString()
+        Log.e("num_people_movies",num_people_movies)
```
Comment (Saif-Alhaider) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsViewModel.kt:82]: remove log

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsViewModel.kt [hunk 33]
```diff
@@ -0,0 +1,130 @@
...
+    private val getMoviesByPeopleUseCase: GetMoviesByPeopleUseCase,
+    private val getTvShowsByPeopleUseCase: GetTvShowsByPeopleUseCase,
+    private val peopleDataUiMapper: PeopleDataUiMapper,
+    private val moviesByPeopleUiMapper: MoviesByPeopleUiMapper,
+    private val tvShowsByPeopleUiMapper: TvShowsByPeopleUiMapper
+
+) : BaseViewModel<PeopleDetailsUiState, PeopleDetailsUiEvent>(PeopleDetailsUiState()),
+    PeopleDetailsListener {
+
+    var num_people_movies = ""
```
Comment (Saif-Alhaider) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsViewModel.kt:33]: avoid var variables in viewmodel

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/mapper/TvShowsByPeopleUiMapper.kt [hunk 34]
```diff
@@ -0,0 +1,17 @@
+package com.chocolatecake.viewmodel.people.mapper
+
+import com.chocolatecake.entities.TvShowsEntity
+import com.chocolatecake.mapper.Mapper
+import com.chocolatecake.viewmodel.common.model.MediaVerticalUIState
+import com.chocolatecake.viewmodel.people.PeopleDetailsUiState
+import javax.inject.Inject
+
+class TvShowsByPeopleUiMapper @Inject constructor() :  Mapper<TvShowsEntity, MediaVerticalUIState> {
```
Comment (Saif-Alhaider) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/mapper/TvShowsByPeopleUiMapper.kt:9]: should map to PeopleDetailsUiState and not MediaVerticalUIState

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/mapper/PeopleDataUiMapper.kt [hunk 35]
```diff
@@ -0,0 +1,23 @@
+package com.chocolatecake.viewmodel.people.mapper
+
+import com.chocolatecake.entities.PeopleDataEntity
+import com.chocolatecake.mapper.Mapper
+import com.chocolatecake.viewmodel.people.PeopleDetailsUiState
+import javax.inject.Inject
+
+class PeopleDataUiMapper @Inject constructor() :
+    Mapper<PeopleDataEntity, PeopleDetailsUiState.PeopleDataUiState> {
```
Comment (Saif-Alhaider) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/mapper/PeopleDataUiMapper.kt:9]: should map to PeopleDetailsUiState and not PeopleDetailsUiState.PeopleDataUiState

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/mapper/MoviesByPeopleUiMapper.kt [hunk 36]
```diff
@@ -0,0 +1,17 @@
+package com.chocolatecake.viewmodel.people.mapper
+
+import com.chocolatecake.entities.MovieEntity
+import com.chocolatecake.mapper.Mapper
+import com.chocolatecake.viewmodel.common.model.MediaVerticalUIState
+import com.chocolatecake.viewmodel.people.PeopleDetailsUiState
+import javax.inject.Inject
+
+class MoviesByPeopleUiMapper @Inject constructor():  Mapper<MovieEntity, MediaVerticalUIState> {
```
Comment (Saif-Alhaider) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/mapper/MoviesByPeopleUiMapper.kt:9]: should map to PeopleDetailsUiState and not MediaVerticalUIState

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsViewModel.kt [hunk 37]
```diff
@@ -0,0 +1,130 @@
...
+        }
+        num_people_movies=moviesUiState.size.toString()
+        Log.e("num_people_movies",num_people_movies)
+    }
+
+    private fun onErrorGetMoviesByPeople(e: Throwable) {
+        val errors = _state.value.onErrors.toMutableList()
+        errors.add(e.message.toString())
+        _state.update { it.copy(onErrors = errors, isLoading = false) }
+    }
```
Comment (Saif-Alhaider) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsViewModel.kt:89]:
```suggestion
    private fun onError(th: Throwable) {
        val errors = _state.value.errors.toMutableList()
        errors.add(th.message.toString())
        _state.update { it.copy(errors = errors, isLoading = false) }
    }
```
most viewModels in the project uses this implementation

**Review by Saif-Alhaider — COMMENTED** [review]
urgent: should refactor

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsViewModel.kt (hunk 37 above)
Comment (Hagar-Gomaa) [reply feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsViewModel.kt:89]: what you mean

Comment (Hagar-Gomaa) [reply feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/people/PeopleDetailsViewModel.kt:89]: what you mean

**Pull request closed without merging**